go/filemanager/pfsmodules/upload.go

```diff
@@ -91,100 +35,69 @@ func getChunkReader(path string, offset int64) (*os.File, error) {
-	cmd := Chunk{
-		Path:   dst,
-		Offset: src.Offset,
-		Size:   src.Size,
-	}
-
-	return cmd.ToURLParam().Encode()
-}
-
-func postChunk(src *Chunk, dst string) ([]byte, error) {
-	f, err := getChunkReader(src.Path, src.Offset)
-	if err != nil {
-		return nil, err
+func uploadFile(src, dst string, srcFileSize int64) error {
+	r := FileHandle{}
+	if err := r.Open(src, os.O_RDONLY, 0); err != nil {
+		return err
 	}
-	defer Close(f)
+	defer r.Close()
 
-	t := fmt.Sprintf("%s/api/v1/pfs/storage/chunks", Config.ActiveConfig.Endpoint)
-	log.V(4).Infoln(t)
-
-	return restclient.PostChunk(t, getDstParam(src, dst),
-		f, src.Size, DefaultMultiPartBoundary)
-}
-
-func uploadChunks(src string,
-	dst string,
-	diffMeta []ChunkMeta) error {
-	if len(diffMeta) == 0 {
-		log.V(1).Infof("srcfile:%s and destfile:%s are same\n", src, dst)
-		return nil
+	w := RFileHandle{}
+	if err := w.Open(dst, os.O_RDWR, srcFileSize); err != nil {
+		return err
 	}
+	defer w.Close()
 
-	for _, meta := range diffMeta {
-		log.V(3).Infof("diffMeta:%v\n", meta)
+	// upload chunks.
+	size := defaultChunkSize
 
-		chunk := Chunk{
-			Path:   src,
-			Offset: meta.Offset,
-			Size:   meta.Len,
+	offset := int64(0)
+	for {
+		log.V(2).Infoln("\n")
+		c, err := r.ReadChunk(offset, size)
+		if err == io.EOF {
+			break
 		}
-
-		body, err := postChunk(&chunk, dst)
 		if err != nil {
 			return err
 		}
+		log.V(2).Infoln("local chunk info:" + c.ToString())
 
-		resp := uploadChunkResponse{}
-		if err := json.Unmarshal(body, &resp); err != nil {
+		m, err := w.GetChunkMeta(offset, size)
+		if err != nil {
 			return err
 		}
+		log.V(2).Infoln("remote chunk info:" + m.ToString())
+		offset += c.Len
 
-		if len(resp.Err) == 0 {
+		if c.Checksum == m.Checksum {
+			log.V(2).Infof("remote and local chunk are same chunk info:%s\n", c.ToString())
 			continue
 		}
 
-		return errors.New(resp.Err)
-	}
-
-	return nil
-}
-
-func uploadFile(src, dst string, srcFileSize int64) error {
-
-	log.V(1).Infof("touch %s size:%d\n", dst, srcFileSize)
-
-	cmd := TouchCmd{
-		Method:   TouchCmdName,
-		Path:     dst,
-		FileSize: srcFileSize,
-	}
-
-	if err := remoteTouch(&cmd); err != nil {
-		return err
+		if err := w.WriteChunk(c); err != nil {
+			return err
+		}
+		log.V(2).Infof("upload chunk:%s ok\n", c.ToString())
 	}
 
-	dstMeta, err := remoteChunkMeta(dst, defaultChunkSize)
-	if err != nil {
-		return err
+	if offset != srcFileSize {
+		return fmt.Errorf("expect %d upload %d", srcFileSize, offset)
 	}
-	log.V(2).Infof("dst %s chunkMeta:%#v\n", dst, dstMeta)
 
-	srcMeta, err := GetChunkMeta(src, defaultChunkSize)
-	if err != nil {
-		return err
-	}
-	log.V(2).Infof("src %s chunkMeta:%#v\n", src, srcMeta)
+	return nil
+}
 
-	diffMeta, err := GetDiffChunkMeta(srcMeta, dstMeta)
-	if err != nil {
-		return err
-	}
-	log.V(2).Infof("diff chunkMeta:%#v\n", diffMeta)
+// ColorError print red ERROR before message.
+func ColorError(format string, a ...interface{}) {
```

Comment:
We can consider changing to `https://github.com/sirupsen/logrus`, it supports colored `[ERROR]` message by default.